### PR TITLE
cdxgen: 10.5.2 -> 10.8.1

### PR DESCRIPTION
**Pull request by Quince-Pie**
## Description of changes

Changelogs: 
- https://github.com/CycloneDX/cdxgen/releases/tag/v10.8.1
- https://github.com/CycloneDX/cdxgen/releases/tag/v10.8.0
- https://github.com/CycloneDX/cdxgen/releases/tag/v10.7.0
- https://github.com/CycloneDX/cdxgen/releases/tag/v10.6.2
- https://github.com/CycloneDX/cdxgen/releases/tag/v10.6.1
- https://github.com/CycloneDX/cdxgen/releases/tag/v10.6.0

Notably: "Switch to pnpm" which removes the package-lock file. 

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/tools/security/cdxgen/default.nix [hunk 1]
```diff
@@ -1,26 +1,66 @@
-{ lib
-, fetchFromGitHub
-, buildNpmPackage
+{ fetchFromGitHub
+, lib
+, makeWrapper
+, nodejs
+, nodePackages
+, pnpm
```
Comment (dit7ya) [inline pkgs/tools/security/cdxgen/default.nix:6]: I think it has been customary to specify it as `pnpm_9` in existing packages.

File: pkgs/tools/security/cdxgen/default.nix [hunk 2]
```diff
@@ -1,26 +1,66 @@
...
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/bin $out/lib
+    cp -r * $out/lib
+    makeWrapper ${nodejs}/bin/node "$out/bin/cdxgen" --add-flags "$out/lib/bin/cdxgen.js"
+    makeWrapper ${nodejs}/bin/node "$out/bin/evinse" --add-flags "$out/lib/bin/evinse.js"
+    makeWrapper ${nodejs}/bin/node "$out/bin/repl" --add-flags "$out/lib/bin/repl.js"
+    makeWrapper ${nodejs}/bin/node "$out/bin/verify" --add-flags "$out/lib/bin/verify.js"
```
Comment (dit7ya) [inline pkgs/tools/security/cdxgen/default.nix:54]: I am not sure what the other binaries do. But I would name them like `cdxgen-evinse` etc.

File: pkgs/tools/security/cdxgen/default.nix [hunk 3]
```diff
@@ -1,26 +1,66 @@
...
+, nodejs
+, nodePackages
+, pnpm
+, python3
+, stdenv
+, xcbuild
 }:
 
-buildNpmPackage rec {
+stdenv.mkDerivation rec {
```
Comment (dit7ya) [inline pkgs/tools/security/cdxgen/default.nix:12]: I am not sure if something changed recently, but last time I checked, we were using the `finalAttrs` pattern for `stdenv.mkDerivation`.

**Review by dit7ya — CHANGES_REQUESTED** [review]
Thanks for the PR, I have left a few comments.

**Comment by Quince-Pie** [comment]
Result of `nixpkgs-review pr 326449` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>cdxgen</li>
  </ul>
</details>

**Review by dit7ya — APPROVED** [review]
LGTM, thanks!

**Review by SuperSandro2000 — APPROVED** [review]

**Pull request merged**